src/libs/Notification/PushNotification/NotificationType.ts

```diff
@@ -0,0 +1,28 @@
+import {OnyxServerUpdate} from '@src/types/onyx/OnyxUpdatesFromServer';
+
+const NotificationType = {
+    REPORT_COMMENT: 'reportComment',
+} as const;
+
+type NotificationDataMap = {
+    [NotificationType.REPORT_COMMENT]: ReportCommentNotificationData;
+};
+
+type NotificationData = ReportCommentNotificationData;
+
+type ReportCommentNotificationData = {
+    title?: string;
+    type?: typeof NotificationType.REPORT_COMMENT;
+    reportID?: number;
+    reportActionID?: string;
```

Comment:
You can mark title, type, reportID and reportActionID is not optional here. These should always be set on the backend.

Reply:
I left `type` and `reportID` as optional because of evidences [here](https://github.com/fabioh8010/expensify-app/blob/ts/lib/PushNotification/src/libs/Notification/PushNotification/index.native.ts#L43), [here](https://github.com/fabioh8010/expensify-app/blob/ts/lib/PushNotification/src/libs/Notification/PushNotification/shouldShowPushNotification.ts#L22) and [here](https://github.com/fabioh8010/expensify-app/blob/ts/lib/PushNotification/src/libs/Notification/PushNotification/subscribeToReportCommentPushNotifications.ts#L21).

So I ended up leaving everything as optional. @arosiclair @Julesssss  Could you confirm in the backend which properties are optional or required?

Reply:
For now, only onyxData is optional...  but this could change at any time.

Reply:
> I left type and reportID as optional because of evidences [here](https://github.com/fabioh8010/expensify-app/blob/ts/lib/PushNotification/src/libs/Notification/PushNotification/index.native.ts#L43), [here](https://github.com/fabioh8010/expensify-app/blob/ts/lib/PushNotification/src/libs/Notification/PushNotification/shouldShowPushNotification.ts#L22) and [here](https://github.com/fabioh8010/expensify-app/blob/ts/lib/PushNotification/src/libs/Notification/PushNotification/subscribeToReportCommentPushNotifications.ts#L21).

Hmm good finds but these are essentially dead code to be honest. Right now our only type of push notifications are report comment notifications and those definitely need to have the `reportID`, `reportActionID` and `type` set. If they don't that's just a bug on the backend that would need to get fixed ASAP. 

Let's still mark those as required for now and if the schema needs to change down the line we just need to make sure the frontend and backend are updated like usual.